src/App/Fossa/VSI/Types.hs

```diff
@@ -24,6 +27,27 @@ import Effect.Logger (Pretty (pretty), viaShow)
 import Srclib.Converter (depTypeToFetcher, fetcherToDepType)
 import Srclib.Types qualified as Srclib
 
+-- | The VSI backend returns a scan ID when a scan is created, which is then used to add files to the scan and get inferred OSS dependencies.
+newtype ScanID = ScanID {unScanID :: Text} deriving (ToJSON, FromJSON)
+
+instance ToText ScanID where
+  toText = unScanID
```

Comment:
I think we should avoid creating `ToText` instances for `Text`-wrapping newtypes.  It kind of breaks the semantic safety that they offer.  See my comment on `vsiAddFilesToScanEndpoint` for an example.